src/core/expression/qgsexpression.h

```diff
@@ -115,6 +115,49 @@ class CORE_EXPORT QgsExpression
     Q_DECLARE_TR_FUNCTIONS( QgsExpression )
   public:
 
+    /**
+     * Details about any parser errors that were found when parsing the expression.
+     * \since QGIS 3.0
+     */
+    struct CORE_EXPORT ParserError
+    {
+      enum ParserErrorType
+      {
+        Unknown = 0,
+        FunctionUnknown = 1,
```

Comment:
Can you please add docstrings for these values too?

Reply:
Yep.